iot_config/src/org.rs

```diff
@@ -138,7 +149,255 @@ pub async fn create_org(
     Ok(org)
 }
 
-async fn insert_constraints(
+pub async fn update_org(
+    oui: u64,
+    authorizer: UpdateAuthorizer,
+    updates: Vec<proto::UpdateV1>,
+    db: impl sqlx::PgExecutor<'_> + sqlx::Acquire<'_, Database = sqlx::Postgres>,
+) -> Result<Org, OrgStoreError> {
+    let mut txn = db.begin().await?;
+
+    let net_id = get_org_netid(oui, &mut txn).await?;
+    let is_helium_org = is_helium_netid(&net_id);
+    let current_org = get(oui, &mut txn).await?;
+
+    for update in updates {
+        match update.update {
+            Some(proto::Update::Owner(pubkeybin)) if authorizer == UpdateAuthorizer::Admin => {
+                update_owner(oui, pubkeybin.into(), &mut txn).await?
+            }
+            Some(proto::Update::Devaddrs(addr_count))
+                if authorizer == UpdateAuthorizer::Admin && is_helium_org =>
+            {
+                add_devaddr_slab(oui, net_id, addr_count, &mut txn).await?
+            }
+            Some(proto::Update::Constraint(constraint_update))
+                if authorizer == UpdateAuthorizer::Admin && is_helium_org =>
+            {
+                match (constraint_update.action(), &constraint_update.constraint) {
+                    (proto::ActionV1::Add, Some(ref constraint)) => add_constraint_update(oui, net_id, constraint.into(), &mut txn).await?,
+                    (proto::ActionV1::Remove, Some(ref constraint)) => remove_constraint_update(oui, net_id, current_org.constraints.as_ref(), constraint.into(), &mut txn).await?,
+                    _ => return Err(OrgStoreError::InvalidUpdate(format!("invalid action or missing devaddr constraint update: {constraint_update:?}")))
+                }
+            }
+            Some(proto::Update::Payer(pubkeybin)) => {
+                update_payer(oui, pubkeybin.into(), &mut txn).await?
+            }
+            Some(proto::Update::DelegateKey(delegate_key_update)) => {
+                match delegate_key_update.action() {
+                    proto::ActionV1::Add => {
+                        add_delegate_key(oui, delegate_key_update.delegate_key.into(), &mut txn)
+                            .await?
+                    }
+                    proto::ActionV1::Remove => {
+                        remove_delegate_key(oui, delegate_key_update.delegate_key.into(), &mut txn)
+                            .await?
+                    }
+                }
+            }
+            _ => {
+                return Err(OrgStoreError::InvalidUpdate(format!(
+                    "update: {update:?}, authorizer: {authorizer:?}"
+                )))
+            }
+        };
+    }
+
+    let updated_org = get(oui, &mut txn).await?;
+
+    txn.commit().await?;
+
+    Ok(updated_org)
+}
+
+pub async fn get_org_netid(
+    oui: u64,
+    db: impl sqlx::PgExecutor<'_>,
+) -> Result<NetIdField, sqlx::Error> {
+    let netid = sqlx::query_scalar::<_, i32>(
+        " select net_id from organization_devaddr_constraints where oui = $1 limit 1 ",
```

Comment:
using limit without order by here, what about side effects from potential inconsistent results ?   Or is net_id guaranteed to be a consistent value here ?